dockstore-webservice/src/main/java/io/dockstore/webservice/languages/LanguageHandlerInterface.java

```diff
@@ -609,11 +634,159 @@ static DockerSpecifier determineImageSpecifier(String image, DockerImageReferenc
                     dockerImages.addAll(dockerHubImages);
                 }
+            } else if (registryFound == Registry.GITHUB_CONTAINER_REGISTRY) {
+                try {
+                    splitDocker = image.split("/"); // ghcr.io/<repo>/<image>:1 -> ["ghcr.io", "repo-name", "image-name:1"]
+                    String imageNameWithSpecifier = splitDocker[2];
+                    splitSpecifier = imageNameWithSpecifier.split(specifierSymbol); // ["image-name", "1"]
+                } catch (ArrayIndexOutOfBoundsException ex) {
+                    LOG.error("URL to image on GitHub Container Registry incomplete", ex);
+                    continue;
+                }
+
+                if (splitDocker.length > 2) { // ["ghcr.io", "repo-name", "image-name"]
+                    String repo = String.join("/", splitDocker[1], splitSpecifier[0]);
+                    String specifierName = splitSpecifier[1];
+                    Set<Image> gitHubContainerRegistryImages = getImagesFromGitHubContainerRegistry(repo, specifierName, imageSpecifier, client);
+                    dockerImages.addAll(gitHubContainerRegistryImages);
+                } else {
+                    LOG.error("Could not find image version specified for {}", splitDocker[1]);
+                }
             }
         }
         return dockerImages;
     }
 
+    /**
+     * Create an Image by retrieving image metadata, including the checksum, using the Docker Registry HTTP API V2.
+     * A work-around solution to harvest checksums for images belonging to registries that require authentication to their API.
+     *
+     * @param registry
+     * @param repo
+     * @param specifierName
+     * @param specifierType
+     * @param client
+     * @return
+     */
+    default Optional<Image> getImage(Registry registry, String repo, String specifierName,
+            DockerSpecifier specifierType, HttpClient client) {
+        String getTokenURL = String.format("https://%s/token?scope=repository:%s:pull&service=%s",
+                registry.getDockerPath(), repo, registry.getDockerPath());
+        // Ex: https://ghcr.io/v2/<repo>/manifests/<tag_or_digest>
+        String getImageManifestURL = String.format("https://%s/v2/%s/manifests/%s", registry.getDockerPath(), repo, specifierName);
+        // Ex: https://ghcr.io/v2/<repo>/blobs/<digest>
+        String getBlobURLFormat = "https://%s/v2/%s/blobs/%s";
+        String acceptSchema2ManifestHeader = "application/vnd.docker.distribution.manifest.v2+json";
+        HttpGet httpRequest = new HttpGet(getImageManifestURL);
+
+        try {
+            // Get token with pull access to make Docker Registry HTTP API V2 calls.
+            // Source for token request specs: https://docs.docker.com/registry/spec/auth/token/#requesting-a-token
+            Optional<String> tokenResponse = ResourceUtilities.asString(getTokenURL, null, client);
```

Comment:
Interesting that you can anonymously get a token.

Reply:
Interesting, any idea what the usage limits are on an anonymous token? 
Wondering how often we might hit rate limit with this when multiple people are using this